#!/usr/bin/env python

"""
This config file defines the parameters of the preprocessing
nose(preprocessing_node.cpp) adjustable using the dynamic reconfigure module.
"""

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "ugoe_gap_detection_ros"
gen = ParameterGenerator()

so_group = gen.add_group("Statistical_Outlier_Denoising")
so_group.add("MeanK",                        int_t,    0,
             "Number of nearest neighbors to use for mean distance estimation",
             150, 0, 300)
so_group.add("StdDevMulThresh",           double_t,    0,
             "The standard deviation multiplier for the distance threshold " +
             "calculation",
             30.0, 0.0, 30.0)

roi_group = gen.add_group("Region_of_Interest_selection")
roi_group.add("zLowerLimit",      double_t,    0,
              "Lower filter bound for the z-Axis", -0.30,  -0.30, 0.0)
roi_group.add("zUpperLimit",      double_t,    0,
              "Upper filter bound for the z-Axis", 0.30,  0.0, 0.30)
roi_group.add("yLowerLimit",      double_t,    0,
              "Lower filter bound for the y-Axis", -0.30,  -0.30, 0.0)
roi_group.add("yUpperLimit",      double_t,    0,
              "Upper filter bound for the y-Axis", 0.30,  0.0, 0.30)
roi_group.add("xLowerLimit",      double_t,    0,
              "Lower filter bound for the x-Axis", 0.0,  0.0, 1.0)
roi_group.add("xUpperLimit",      double_t,    0,
              "Upper filter bound for the x-Axis", 1.0,  0.0, 1.0)

gen.add("create_PCD",   bool_t,   0, "Create a denoised PCD file for " +
        "evaluation",  False)

exit(gen.generate(PACKAGE, "preprocessing_node", "preprocessing"))
